### DISCUSSION TITLE
"Restriction" mods category

### DISCUSSION THREAD

**Discussion by peppy** [ppy/osu#20056] [Ideas]
As discussed in conversations with multiple people recently, a new category of mods will do us well. Opening this issue to keep track of the mods I'd see being added. Feel free to reply with any suggestions which could fit into this category, with the criteria being that:

- The mods should not affect gameplay directly
- The mods can lock game settings, or change how score submissions behaves
- The mods should be seen as "wow" factor. Having them on your score should increase its perceived value, even if there's no multiplier applied (I would see the mod icons having a special flair and potentially position in the result screen).

"[Overwrite](https://github.com/ppy/osu/discussions/20004)" – forcefully overwrite any existing score on the leaderboard for same mod combination (minus the overwrite mod)
"[Committed](https://github.com/ppy/osu/discussions/19848)" – can't quit, retry or pause
"Vibrant" – forced full beatmap skin, storyboard, no background dim (as discussed with @pishifat, could be used for playlists where storyboards are being showcased as one example)

**Comment by Moltanicaa** [comment]


**Reply by peppy** [reply]
This is not relevant in any way, sorry. Please start a separate discussion.

**Comment by iSimpFor** [comment]
sudden death and perfect should 100% be moved into this new category

**Reply by peppy** [reply]
Potentially, yes.

**Comment by Nepp0** [comment]
idea: mod that randomizes sensitivity or area within the selected range

**Reply by uzervlad** [reply]
Doesn't feel like a "restriction" mod to me

**Reply by Nepp0** [reply]
it fits the criteria, i think it would be a cool addition.

**Reply by uzervlad** [reply]
I'd argue that it doesn't, since it directly affects gameplay by changing the controls, which goes against the first criteria

**Reply by Nepp0** [reply]
you could also argue that "vibrant" directly affects gameplay by changing background dim.
the "wow" factor in both of them is that opacity and sensitivity are both settings that not many people are willing to change, this mod could be a fun challenge, it changes a specific setting which doesnt directly affect gameplay like other mods in the "fun" category do.

**Reply by uzervlad** [reply]
In my understanding, changing how the game looks is not directly affecting gameplay, while how the game is played is.

**Reply by Doksha** [reply]
yeah this is definitely not a restriction mod. This would be a 'fun' mod.
This directly affects how your input is registered, restriction mods should not change anything about gameplay if you follow the rules. Overwrite only affects scores after they are submitted, it has nothing to do with gameplay. Committed prevents you from leaving a map, but doesnt affect anything if you dont try to leave the map. SD and PF kill you if you miss/100 but they dont do anything if you FC/SS. Vibrant changes the background dim and sets storyboard on, but doesnt do anything if those settings are already set.

**Reply by Nepp0** [reply]
SD/PF and vibrant do affect gameplay directly, of course some people's settings and accuracy could match the mod and it wouldnt affect them, but it is unlikely to happen in most cases. same can be said with sensitivity randomize mod. it just forces you do adapt to different cursor speeds, and your cursor sensitivity could match the randomized one but it is very unlikely to happen (maybe make the rate change only after you beat the map?), and of course like SD/PF it wouldnt affect anything if you just play the map properly, just aiming and clicking circles. nothing too groundbreaking here.

**Reply by Doksha** [reply]
Your reasoning makes absolutely no sense. A sensitivity mod would (effectively) shrink and grow the playfield (not literally, but has the same effect) this DOES change how you play, it changes what you have to input for the same output. A successful playthrough with OW/CM/PF/SD/AL/ST/VB will be the EXACT same as a playthough without those mods. If you took a replay with any of those mods, it would work perfectly with those mods removed and (while you may need to change dim and SB for vibrant) will look identical to a play without those mods.

The "Restriction" mods category is *specifically* for mods that simply act as "tokens" of how the map was played, and have no bearing on actual difficulty.

**Reply by peppy** [reply]
No this is stupid. It's not restricting anyting. Please refrain from posting in this thread.

**Comment by mk56-spn** [comment]
i think singletap and alternate should be moved to this. singletap especially is ridiculous  on anything with any high bpm streams.

i think a mod, if you can even call it that, that forces you through the entire mapset from easiest diff to top as one continous map would be very fun, competing to see who could ace a mapset first, or get lowest combined 100s or whatever it is, but again im not sure thats even doable as a mod since its larger in scope that a single beatmap

**Reply by Doksha** [reply]
Maybe committed on a playlist would force you to play every map in a row.

**Comment by peppy** [comment]
Locking thread because the point is being missed. It was mainly meant for internal notetaking anyway.
